ecs-cli/modules/aws/clients/ecs/client.go

```diff
@@ -133,14 +133,28 @@ func (client *ecsClient) DeleteService(serviceName string) error {
 	return nil
 }
 
-func (client *ecsClient) CreateService(serviceName, taskDefName string, deploymentConfig *ecs.DeploymentConfiguration) error {
-	_, err := client.client.CreateService(&ecs.CreateServiceInput{
-		DesiredCount:            aws.Int64(0),            // Required
-		ServiceName:             aws.String(serviceName), // Required
-		TaskDefinition:          aws.String(taskDefName), // Required
-		Cluster:                 aws.String(client.params.Cluster),
-		DeploymentConfiguration: deploymentConfig,
-	})
+func (client *ecsClient) CreateService(serviceName, taskDefName string, loadBalancer *ecs.LoadBalancer, role string, deploymentConfig *ecs.DeploymentConfiguration) error {
+	var err error
+
+	if *loadBalancer.TargetGroupArn != "" {
```

Comment:
As per `loadContext` method https://github.com/aws/amazon-ecs-cli/pull/211/files#diff-97b4106c54aad3d402f437209e9fc4e3R78 `loadBalancer` could have empty values but it'll never be `nil`.